pkg/config/spec.go

```diff
@@ -292,7 +292,7 @@ func setUpgradeSourceSize(cfg *Config, spec *v1.UpgradeSpec) error {
 		targetSpec = &(spec.Active)
 	}
 
-	if targetSpec.Source.IsEmpty() {
+	if targetSpec.Source == nil || targetSpec.Source.IsEmpty() {
```

Comment:
this is extra but nice to have. Found out that if we reach here with an empty source, source will be nil and the agent will panic. So better to have that check in there. Should not happen in mainline as we got checks for that before on the spec.Sanitize()

Reply:
maybe we should have a function for it with a comment, or just the comment, because your explanation makes it clear, but if I just read the code I wouldn't know this is the case

Reply:
Good point, will add a comment